tests/unit/package_managers/yarn/test_main.py

```diff
@@ -103,25 +102,6 @@ def test_configure_yarn_version_fail(
         _configure_yarn_version(mock_project)
 
 
-@mock.patch("cachi2.core.package_managers.yarn.main.run_yarn_cmd")
-def test_fetch_dependencies(mock_yarn_cmd: mock.Mock, rooted_tmp_path: RootedPath) -> None:
-    source_dir = rooted_tmp_path
-    output_dir = rooted_tmp_path.join_within_root("cachi2-output")
-
-    mock_yarn_cmd.side_effect = YarnCommandError("berryscary")
-
-    with pytest.raises(YarnCommandError) as exc_info:
-        _fetch_dependencies(source_dir, output_dir)
-
-    mock_yarn_cmd.assert_called_once_with(
-        ["install", "--mode", "skip-build"],
```

Comment:
One of the things this test was checking was that `run_yarn_cmd` was being called with the right parameters. I think this could be moved to `test_resolve_yarn_project`, but this test does not exist currently.

We should do a follow up PR to increase the coverage on the yarn/main module: https://coveralls.io/builds/64062886/source?filename=cachi2%2Fcore%2Fpackage_managers%2Fyarn%2Fmain.py#L51

Reply:
There's a fine line where the code coverage is a nice thing and where it leads to duplicate test cases IMO. At first I didn't see your point, but now I see you're specifically interested in testing a given set of call parameters which wouldn't be tested anywhere anymore. I'll simplify the current test and keep it around instead of removing it - we can schedule it for a conversion to the `test_resolve_yarn_project` as you noted in a follow up PR. However, as I said, we need to be careful with the test cases we test, since in this particular case we're talking about higher level helpers which transitively exercise functionality of the elementary helper primitives for which we have like 99.8% coverage of various inputs, so we'd only be creating redundancy with an extensive parametrization of say `_fetch_dependencies` or future `test_resolve_yarn_project`. Anyways, for now, I'll keep things as they are :).